backend/collect/utils.py

```diff
@@ -0,0 +1,29 @@
+from django.conf import settings
+from rdflib import RDF, URIRef, Graph
+import re
+
+from triplestore.constants import EDPOPCOL
+
+def _name_to_slug(name: str) -> str:
+    no_spaces = lambda s: re.sub(r'\s+', '_', s)
+    clean = lambda s: re.sub(r'[^a-z0-9\-_]', '', s)
+    strip_start = lambda s: re.sub(r'^\W+', '', s)
+    strip_end = lambda s: re.sub(r'\W+$', '', s)
+
+    return strip_end(strip_start(clean(no_spaces(str.lower(name)))))
+
+
+def collection_uri(name: str):
+    id = _name_to_slug(name)
+    return URIRef(settings.RDF_NAMESPACE_ROOT + 'collections/' + id)
```

Comment:
This constructs the URI for a collection based on the "human" name, i.e. "My collection" will be stored as `my_collection`. This simplifies the frontend UI, which is nice for an initial version, though it may lead to some annoyances long-term.

Reply:
What annoyances?

Reply:
URIs need to be unique, so the request will fail if the URI already exists. This isn't really a problem right now, but if a group of people is enthusiastically adding data on the same topics, conflicts may become more common.

I think there are some minor annoyances that can pop up if you get these warnings:

- You can get a response that the URI is taken while the existing collection is privately owned by another user. This makes technical sense, but it feels odd.
- Names don't need to be unique but URIs do, so this can feel un-intuitive. For instance, if "My collection" / `my_collection` exists, you can get the name you want by creating "My collection 2" / `my_collection_2`, and then renaming it to "My collection" (which will preserve the URI `my_collection_2`).
- "my collection" and "My Collection :)" would both be encoded as `my_collection`, so the user may feel like they don't have full control over the URI. Which is annoying if their collection gets rejected on the basis of the URI.

Reply:
Maybe number slugs automatically, then?